Products/CMFPlone/utils.py

```diff
@@ -234,15 +237,13 @@ def getEmptyTitle(context, translated=True):
 
 
 def typesToList(context):
-    ntp = getToolByName(context, 'portal_properties').navtree_properties
-    ttool = getToolByName(context, 'portal_types')
-    bl = ntp.getProperty('metaTypesNotToList', ())
-    bl_dict = {}
-    for t in bl:
-        bl_dict[t] = 1
-    all_types = ttool.listContentTypes()
-    wl = [t for t in all_types if not t in bl_dict]
-    return wl
+    registry = getUtility(IRegistry)
+    from Products.CMFPlone.interfaces import INavigationSchema
```

Comment:
We are about to prepare the beta2 release. I will merge this pr now to get the fix in. We can work on that later. Can you point me to the plone.app.z3cform validators?